web/source/kmwosk.js

```diff
@@ -763,12 +763,6 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
         // Turn off key highlighting (or preview)
         osk.highlightKey(e,false);
 
-        // Process modifier key action
-        // TODO: Fixup the duplication of nextlayer processing here and below.
-        //       This breaks the `nextlayer` property on keys with output
-        //       even though it does the right thing for modifier keys.
-        if(osk.selectLayer(keyName,nextLayer)) return true;
```

Comment:
Yup.  Turns out the 'duplication of nextlayer' below here was wholly unnecessary, as the code toward the bottom of this function handles all next layer processing quite safely.

There is the question about why this line was here in the first place, admittedly; possibly an artifact leftover after some refactoring?

Reply:
Question answered - it's for desktop click-based OSK interaction.